### PR TITLE
Add Python 3.9 to test matrix

### PR DESCRIPTION
**Pull request by thisac**
**Current issue with Torch 1.7.1 being needed for Python 3.9, although tests randomly crash (unexpectedly) mid-tests, or sometimes in other places, with that version. Otherwise tests seem to pass, and it works locally.**

**Context:**
Adds Python 3.9 tests to the CI.

**Description of the Change:**
Adds Python 3.9 tests to the CI, together with PyTorch and Jax (TensorFlow doesn't support Python 3.9 yet).

**Benefits:**
Tests are run with Python 3.9.

**Possible Drawbacks:**
None

**Related GitHub Issues:**
None

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ jobs:
           - {python-version: 3.7, interfaces: ['torch']}
           - {python-version: 3.8, interfaces: ['tf', 'torch', 'jax']}
           - {python-version: 3.8, interfaces: []}
+          - {python-version: 3.9, interfaces: [torch', 'jax']}
```
Comment (Thenerdstation) [inline .github/workflows/tests.yml:26]:
```suggestion
          - {python-version: 3.9, interfaces: ['torch', 'jax']}

```

**Review by josh146 — APPROVED** [review]
Thanks @thisac!

Can I get you to also increment `after_n_builds` in the `codecov.yml` file? This tells CodeCov not to compute the coverage percentage until it has received `n` build logs.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (thisac) [reply .github/workflows/tests.yml]: Oh, good catch @Thenerdstation!

**Comment by josh146** [comment]
@thisac I just merged in master - a lot more packages now have 3.9 support, so I'm curious if the CI will now pass for pytorch :)

**Comment by josh146** [comment]
Closing this for now, as it has become superseded by newer PRs.

**Pull request closed without merging**